public/app/features/variables/adhoc/actions.ts

```diff
@@ -85,27 +85,16 @@ export const changeVariableDatasource = (datasource?: DataSourceRef): ThunkResul
   return async (dispatch, getState) => {
     const { editor } = getState().templating;
     const variable = getVariable(editor.id, getState());
-
-    const loadingText = 'Ad hoc filters are applied automatically to all queries that target this data source';
-
-    dispatch(
-      changeVariableEditorExtended({
-        propName: 'infoText',
-        propValue: loadingText,
-      })
-    );
     dispatch(changeVariableProp(toVariablePayload(variable, { propName: 'datasource', propValue: datasource })));
 
     const ds = await getDatasourceSrv().get(datasource);
 
-    if (!ds || !ds.getTagKeys) {
-      dispatch(
-        changeVariableEditorExtended({
-          propName: 'infoText',
-          propValue: 'This data source does not support ad hoc filters yet.',
-        })
-      );
-    }
+    // TS TODO: ds is not typed to be optional - is this check unnecessary or is the type incorrect?
+    const message = ds?.getTagKeys
+      ? 'Ad hoc filters are applied automatically to all queries that target this data source'
+      : 'This data source does not support ad hoc filters yet.';
+
+    dispatch(changeVariableEditorExtended({ infoText: message }));
```

Comment:
I've changed the behavior of the action here to no longer dispatch the "Ad hoc filters are applied automatically..." message while the datasource is loading because a) i don't think that's actually a loading messages, and b) I think it only shows _very_ briefly for unsupported data sources anyway?